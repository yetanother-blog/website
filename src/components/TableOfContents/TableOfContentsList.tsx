import React from 'react';
import { GithubSlugger } from 'github-slugger-typescript';
import { Link } from '../Link/Link';
import { Link as GatsbyLink } from 'gatsby';

interface HeadingsProps {
  value: string;
  depth: number;
}

interface TableOfContentsProps {
  headings: [HeadingsProps];
}

const slugger = new GithubSlugger();

export const TableOfContentsList: React.FC<TableOfContentsProps> = (props) => {
  return (
    <ul>
      {props.headings.map((heading) => (
        <li key={heading.value}>
          <Link to={'#' + slugger.slug(heading.value)} as={GatsbyLink}>
            {heading.value}
          </Link>
        </li>
      ))}
    </ul>
  );
};
