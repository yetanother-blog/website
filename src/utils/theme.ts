import { DefaultTheme } from 'styled-components';

export const theme: DefaultTheme = {
  fontSizes: ['12px', '14px', '16px', '20px', '24px', '32px'],
  colors: {
    primary: '#00FF00',
    secondary: '#FB00FF',
    grey500: '#1C1C1C',
    grey400: '#ACADAC',
    grey300: '#F6F6F6',
  },
  breakpoints: ['576px', '768px', '992px', '1200px'],
  space: {
    zero: 0,
    xxs: 4,
    xs: 8,
    s: 12,
    m: 16,
    l: 20,
    xl: 40,
    xxl: 60,
    xxxl: 120,
  },
  radii: {
    s: 3,
  },
  width: {
    m: '620px',
    l: '940px',
    xl: '1200px',
  },
};
