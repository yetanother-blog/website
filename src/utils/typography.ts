import { CSSObject } from "styled-components";
import Typography from "typography";

const typography = new Typography({
  googleFonts: [
    {
      name: "Source Sans Pro",
      styles: ["400", "600"]
    },
    {
      name: "Source Code Pro",
      styles: ["600", "700"]
    }
  ],
  baseFontSize: "20px",
  baseLineHeight: 1.85,
  bodyFontFamily: ["Source Sans Pro", "sans-serif"],
  boldWeight: "600",
  overrideStyles: ({ adjustFontSizeTo }, options, styles) => ({
    a: {
      fontFamily: `"Source Code Pro", "sans-serif"`
    }
  })
});

// Hot reload typography in development.
if (process.env.NODE_ENV !== `production`) {
  typography.injectStyles();
}

export default typography;

export const rhythm = typography.rhythm;
export const scale = typography.scale;

type StyledScale = (values: number) => CSSObject;
export const styledScale = scale as StyledScale;
