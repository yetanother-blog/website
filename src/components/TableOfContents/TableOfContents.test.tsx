import React from 'react';
import { TableOfContentsList } from './TableOfContentsList';
import { render } from '@testing-library/react';
import { ThemeContext } from 'styled-components';
import { theme } from '../../utils/theme';

describe('Table Of Contents ui', () => {
  it('should render toc list w/ items', () => {
    const { container } = render(
      <ThemeContext.Provider value={theme}>
        <TableOfContentsList
          headings={[
            { value: 'some headline', depth: 1 },
            { value: 'some other headline', depth: 1 },
            { value: 'more headlines', depth: 1 },
          ]}
        />
      </ThemeContext.Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
