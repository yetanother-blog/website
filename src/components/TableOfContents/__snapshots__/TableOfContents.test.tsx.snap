// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Of Contents ui should render toc list w/ items 1`] = `
<div>
  <div>
    <h2
      class="sc-iBPRYJ YSVVB"
    >
      Table Of Contents
    </h2>
    <ul
      class="sc-bdfBwQ cclmkS"
    >
      <li
        class="sc-bdfBwQ ivfIwt"
      >
        <a
          class="sc-eCssSg fJLwcK"
          href="/#some-headline"
          type="inline"
        >
          <span
            class="sc-gsTCUz iPjDLQ"
          >
            some headline
          </span>
        </a>
      </li>
      <li
        class="sc-bdfBwQ ivfIwt"
      >
        <a
          class="sc-eCssSg fJLwcK"
          href="/#some-other-headline"
          type="inline"
        >
          <span
            class="sc-gsTCUz iPjDLQ"
          >
            some other headline
          </span>
        </a>
      </li>
      <li
        class="sc-bdfBwQ ivfIwt"
      >
        <a
          class="sc-eCssSg fJLwcK"
          href="/#more-headlines"
          type="inline"
        >
          <span
            class="sc-gsTCUz iPjDLQ"
          >
            more headlines
          </span>
        </a>
      </li>
    </ul>
  </div>
</div>
`;
